pkg/conditions/conditions.go

```diff
@@ -8,3 +8,5 @@ const (
 	// BindingReady indicates that the binding succeeded
 	BindingReady conditionsv1.ConditionType = "Ready"
 )
+
+const EmptyServiceSelectorsReason = "EmptyServiceSelectors"
```

Comment:
I consider conditions to be a domain worth of its own package. Additionally, it provides the separation to avoid cyclic dependencies issues in the operator's import tree, should we choose to continue to further refactor the operator internals.

Reply:
Let's try not to look too far in the future, let's keep it as simple as possible, and avoid adding public APIs if not needed now. If `pkg/controller/servicebindingrequest` is only place where we use it, let's move them there and make them private. If we later see that we need to introduce any public API, we can do it. It is a way easier to introduce something than to deprecate it. 

I tried to move these constants locally and did not see any cyclic dependency issues, the code compiled.

Reply:
If that's the only complain about this, let's please address it in a following PR.

Reply:
Why not solving it right here? The constant is introduced in this PR- it did not exist before. Why would we merge it, despite the detected concerns, and then try to fix it later? Reviews should detect issues and solved them (if possible) in the reviewed PR.

Reply:
Because this honestly seems to be a matter of preference: you're advocating to keep all symbols inside the same package implying that this approach is simpler whereas I advocate the usage of different packages to keep symbols belonging to different domains isolated to have conceptual components at hand to increase cohesion and decrease coupling.

Reply:
Right now, we do not have the usage for what you are targeting, at least not in this PR. I am proposing that we gradually evolve the code. If a symbols is only used by a single package, then keep it inside and make it private. If we later need to expose to others, we can make it public and move it to an appropriate place. There is no strong reason to make it public now - that decision can be made once we continue with further refactoring. It is not matter of simpler vs. complex - it is about postponing the decision if something need to public later, when we potentially will have more infos to decide properly.

Reply:
I understand what you're saying, and still consider it a matter of personal preference. 

I have no issue whatsoever with developers introducing a package to separate concerns, and I professionally consider this a good practice.

Please address your concern in a subsequent PR if you think this is worth pursuing.

Reply:
> I understand what you're saying, and still consider it a matter of personal preference.

It is not about the personal preference - reviews are here to compare/discover pros/cons/advantages/disadvantages of code introduced in PR. I did not hear from you what are the benefits of introducing a package with only two constants **NOW**. I am saying that we can postpone creation of such package for later, when we will anyway know better. I am all for separation of concerns and good encapsulation. However, I do not see enough reasons for introducing this package in this PR (as the code shows).

Hence, please explain what is the benefit of introducing package with only two constants in this PR? How does this impact the rest of the code introduced in this PR? If we the move constants elsewhere in this PR, how the rest of the code will be impacted?

Reply:
I don't necessarily think it's a matter of preference tbh. 
As an example, if we generated godocs for this, how would this look ? Are we exposing things that don't need to be exposed at all?

Reply:
Review symbols visibility #462